### DISCUSSION TITLE
fail2ban stopped banning

### DISCUSSION THREAD

**Discussion by chriskuku** [fail2ban/fail2ban#3614] [General]
I'm running an Ubuntu 22.04 server. I see no bans in 3 days now, although fail2ban is running. It is programmed to do
ufw actions:

```
banaction = ufw
banaction_allports = iptables-allports
```

How can I debug fail2ban? Or increase verbosity level? In ufw.log I see many accesses from one IP address in 20s intervals to random ports : 
```
Oct 24 17:19:46 mail kernel: [88762.098932] [UFW BLOCK] IN=eth0 OUT= SRC=84.54.51.91 DST=myip LEN=40 TOS=0x00 PREC=0x00 TTL=245 ID=6912 PROTO=TCP SPT=59017 DPT=10116
Oct 24 17:20:05 mail kernel: [88781.139106] [UFW BLOCK] IN=eth0 OUT= SRC=84.54.51.91 DST=myip LEN=40 TOS=0x00 PREC=0x00 TTL=245 ID=44251 PROTO=TCP SPT=59017 DPT=25440
Oct 24 17:20:25 mail kernel: [88801.029847] [UFW BLOCK] IN=eth0 OUT= SRC=84.54.51.91 DST=myip LEN=40 TOS=0x00 PREC=0x00 TTL=245 ID=41562 PROTO=TCP SPT=59017 DPT=25026
Oct 24 17:21:27 mail kernel: [88862.546265] [UFW BLOCK] IN=eth0 OUT= SRC=84.54.51.91 DST=myip LEN=40 TOS=0x00 PREC=0x00 TTL=245 ID=51085 PROTO=TCP SPT=59017 DPT=5005
Oct 24 17:21:46 mail kernel: [88882.045776] [UFW BLOCK] IN=eth0 OUT= SRC=84.54.51.91 DST=myip LEN=40 TOS=0x00 PREC=0x00 TTL=245 ID=46190 PROTO=TCP SPT=59017 DPT=10061
Oct 24 17:22:06 mail kernel: [88902.072385] [UFW BLOCK] IN=eth0 OUT= SRC=84.54.51.91 DST=myip LEN=40 TOS=0x00 PREC=0x00 TTL=245 ID=17478 PROTO=TCP SPT=59017 DPT=27006
Oct 24 17:22:46 mail kernel: [88942.071817] [UFW BLOCK] IN=eth0 OUT= SRC=84.54.51.91 DST=myip LEN=40 TOS=0x00 PREC=0x00 TTL=245 ID=22217 PROTO=TCP SPT=59017 DPT=10000
Oct 24 17:23:05 mail kernel: [88961.070390] [UFW BLOCK] IN=eth0 OUT= SRC=84.54.51.91 DST=myip LEN=40 TOS=0x00 PREC=0x00 TTL=245 ID=12204 PROTO=TCP SPT=59017 DPT=25944
Oct 24 17:23:25 mail kernel: [88980.908302] [UFW BLOCK] IN=eth0 OUT= SRC=84.54.51.91 DST=myip LEN=40 TOS=0x00 PREC=0x00 TTL=245 ID=39764 PROTO=TCP SPT=59017 DPT=25629
Oct 24 17:23:46 mail kernel: [89001.664958] [UFW BLOCK] IN=eth0 OUT= SRC=84.54.51.91 DST=myip LEN=40 TOS=0x00 PREC=0x00 TTL=245 ID=33083 PROTO=TCP SPT=59017 DPT=10219
Oct 24 17:24:05 mail kernel: [89021.061456] [UFW BLOCK] IN=eth0 OUT= SRC=84.54.51.91 DST=myip LEN=40 TOS=0x00 PREC=0x00 TTL=245 ID=18687 PROTO=TCP SPT=59017 DPT=60006
Oct 24 17:24:45 mail kernel: [89060.984835] [UFW BLOCK] IN=eth0 OUT= SRC=84.54.51.91 DST=myip LEN=40 TOS=0x00 PREC=0x00 TTL=245 ID=9717 PROTO=TCP SPT=59017 DPT=11783
Oct 24 17:25:05 mail kernel: [89081.532995] [UFW BLOCK] IN=eth0 OUT= SRC=84.54.51.91 DST=myip LEN=40 TOS=0x00 PREC=0x00 TTL=245 ID=1526 PROTO=TCP SPT=59017 DPT=25029
Oct 24 17:25:27 mail kernel: [89103.417761] [UFW BLOCK] IN=eth0 OUT= SRC=84.54.51.91 DST=myip LEN=40 TOS=0x00 PREC=0x00 TTL=245 ID=8312 PROTO=TCP SPT=59017 DPT=25240
```
(I trimmed the ufw.log a bit) 
How can I catch these (in case they are candidates to ban at all)

**Comment by sebres** [comment]
> I see no bans in 3 days now, although fail2ban is running. 

Which jails are active (I hope you enabled some of them)? Do you see something related that jails in corresponding logs or journals that must be banned, if so check them with `fail2ban-regex` whether some of the filter matches the log/journal.
Also note [wiki :: How fail2ban works](https://github.com/fail2ban/fail2ban/wiki/How-fail2ban-works), especially the FAQ (first answer).

> How can I debug fail2ban?

You can set loglevel to `DEBUG` (10) or to `HEAVYDEBUG` (5) or even below 5, either in `/etc/fail2ban/fail2ban.local` or via command line option `--loglevel`.
Or even dynamical with `?sudo? fail2ban-client set loglevel HEAVYDEBUG`.
(This will cause fail2ban becomes very wordy, so fail2ban.log would grow very fast, so don't forget to set it back to `INFO`).

If you rather mean the debugging of filter/failregex:
```bash
# check filter over the log:
fail2ban-regex /var/log/ufw.log ufw-port-scan
# check filter over the journal:
fail2ban-regex systemd-journal ufw-port-scan
# check failregex:
fail2ban-regex /var/log/ufw.log '^\s*\S+ kernel:(?: +\[[^\]]+\])? \[UFW BLOCK\](?: (?:IN=\w+|OUT=|(?:(?!OUT=|IN=|SRC=)[A-Z]+=\S*)))* SRC=<ADDR> DST=\S+'
```

> In ufw.log I see many accesses from one IP address in 20s intervals to random ports

This has nothing to do with fail2ban directly. Probably some port scanner or bot crawling your host info.
At least if you don't have a jail, that would block such "attacks" (what is not really one), but it may be too risky (so not advisable till one doesn't really know what one does).
For details see #2225 and https://github.com/fail2ban/fail2ban/issues/1945#issuecomment-362247222

> How can I catch these (in case they are candidates to ban at all)

You need to add a jail with a filter (like in #2225), set logpath and other parameters and enable it. 
But as already said, firstly ensure you understand what you do.
